py/DREAM/Settings/Equations/DistributionFunction.py

```diff
@@ -190,6 +199,49 @@ def setInitialValue(self, f, r, p=None, xi=None, ppar=None, pperp=None):
         self.verifyInitialDistribution()
 
 
+    def setRipple(self, m, n, dB_B, ncoils=0, deltacoils=0, r=[0], t=[0]):
+        """
+        Enable the ripple pitch scattering term.
+
+        :param list m:           Poloidal mode numbers of magnetic perturbation(s).
+        :param list n:           Toroidal mode numbers of magnetic perturbation(s).
+        :param dB_B:             Magnetic perturbations (shape: nModes x nt x nr).
+        :param int ncoils:       Number of toroidal field coils.
+        :param float deltacoils: Distance between toroidal field coils.
+        :param r:                Radial grid on which the magnetic perturbations are given.
+        :param t:                Time grid on which the magnetic perturbations are given.
+        """
+        if type(m) == list: m = np.array(m)
+        elif np.isscalar(m): m = np.array([float(m)])
+
+        if type(n) == list: n = np.array(n)
+        elif np.isscalar(n): n = np.array([float(n)])
+
+        if type(r) == list: r = np.array(r)
+        elif np.isscalar(r): r = np.array([float(r)])
+
+        if type(t) == list: t = np.array(t)
+        elif np.isscalar(t): t = np.array([float(t)])
+
+        if type(dB_B) == list:
+            dB_B = np.array(dB_B)
+        if type(dB_B) == float or dB_B.ndim == 1:
+            dB_B = np.ones((m.size, t.size, r.size)) * dB_B
+
+        if m.size != n.size:
```

Comment:
it would be nice to allow dB_B to be given as a list of length m.size, and in this method rescale it to the correct format as constant in t and r.